R/threshold_Plot.R

```diff
@@ -0,0 +1,191 @@
+      } else {
+        res <- rep(NA, 3)
+      }
+    }
+    return(res)
+  }
+  out <- apply(u.i, 1, thfun,
+               x = x, type = type, a = alpha,
+               verbose = verbose, ...
+  )
+  if (type == "PP") {
+    rownames(out) <- c(
+      "low.loc", "low.scale", "low.shape",
+      "location", "scale", "shape", "up.loc", "up.scale",
+      "up.shape"
+    )
+  } else if (type != "Exponential") {
+    rownames(out) <- c(
+      "low.t.scale", "low.shape", "t.scale",
+      "shape", "up.t.scale", "up.shape"
+    )
+  } else {
+    rownames(out) <- c("low.scale", "scale", "up.scale")
+  }
+  m1 <- deparse(match.call())
+  if (type == "PP") {
+    yl <- range(c(out[c("low.loc", "location", "up.loc"), ]), finite = TRUE)
+    lst_plots[[1]] <- ggplot2::qplot(
+      x = u.i, y = out["location", ], ylim = yl, xlab = xlb,
+      ylab = "location", geom = c("line", "point"), main = main
+    ) +
+      ggplot2::geom_pointrange(mapping = ggplot2::aes(ymin = out["low.loc", ], ymax = out["up.loc", ]))
+    yl <- range(c(out[c("low.scale", "scale", "up.scale"), ]), finite = TRUE)
+    lst_plots[[2]] <- ggplot2::qplot(
+      x = u.i, y = out["scale", ], ylim = yl, xlab = xlb, ylab = "scale",
+      geom = c("point", "line")
+    ) +
+      ggplot2::geom_pointrange(mapping = ggplot2::aes(ymin = out["low.scale", ], ymax = out["up.scale", ]))
+    yl <- range(c(out[c("low.shape", "shape", "up.shape"), ]), finite = TRUE)
+    lst_plots[[3]] <- ggplot2::qplot(u.i, out["shape", ],
+                                     ylim = yl, xlab = "Threshold",
+                                     ylab = "shape", geom = c("point", "line")
+    ) +
+      ggplot2::geom_pointrange(mapping = ggplot2::aes(ymin = out["low.shape", ], ymax = out["up.shape", ]))
+  }
+  else if (type != "Exponential") {
+    yl <- range(c(out[c("low.t.scale", "t.scale", "up.t.scale"), ]), finite = TRUE)
+    lst_plots[[1]] <- ggplot2::qplot(
+      x = u.i, y = out["t.scale", ], ylim = yl, xlab = xlb, ylab = "reparameterized scale",
+      geom = c("point", "line"), main = main) +
+      ggplot2::geom_pointrange(mapping = ggplot2::aes(ymin = out["low.t.scale", ], ymax = out["up.t.scale", ]))
+    yl <- range(c(out[c("low.shape", "shape", "up.shape"), ]), finite = TRUE)
+    lst_plots[[2]] <- ggplot2::qplot(
+      x = u.i, y = out["shape", ], ylim = yl, xlab = "Threshold",
+      ylab = "shape", geom = c("point", "line")
+    ) +
+      ggplot2::geom_pointrange(mapping = ggplot2::aes(ymin = out["low.shape", ], ymax = out["up.shape", ]))
+  }
+  else {
+    yl <- range(c(out[c("low.scale", "scale", "up.scale"), 
+    ]), finite = TRUE)
+    return(ggplot2::qplot(
+      x = u.i, y = out["scale", ], ylim = yl, xlab = "Threshold",
+      ylab = "scale", geom = c("point", "line"), main = main
+    ) +
+      ggplot2::geom_pointrange(mapping = ggplot2::aes(ymin = out["low.scale", ], ymax = out["up.scale", ])))
+  }
+  patchwork::wrap_plots(lst_plots, ncol = 1) 
```

Comment:
We are adding in a new package here - `patchwork` - can you also add this into the DESCRIPTION file?